source/2_platform/rex_opengl/include/rex_opengl/resources/buffer.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include "rex_opengl/resources/resource.h"
+
+#include "rex_engine/types.h"
+
+namespace rex
+{
+    enum class BufferUsage
+    {
+        STATIC_DRAW,
+        DYNAMIC_DRAW
+    };
+
+    class Buffer : public Resource
+    {
+    public:
+        Buffer(u32 target);
+        ~Buffer() override;
```

Comment:
no custom copy and move ctor and assignment provided, making copying possible which you don't want (moving you want perhaps?)